src/Microsoft.PowerShell.Security/security/CredentialCommands.cs

```diff
@@ -54,24 +54,38 @@ public PSCredential Credential
         /// Gets and sets the user supplied message providing description about which script/function is 
         /// requesting the PSCredential from the user.
         /// </summary>
-        [Parameter(Mandatory = true, ParameterSetName = messageSet)]
+        [Parameter(Mandatory = false, ParameterSetName = messageSet)]
+        [ValidateNotNullOrEmpty]
         public string Message
         {
             get { return _message; }
             set { _message = value; }
         }
-        private string _message = null;
+        private string _message;
 
         /// <summary>
         /// Gets and sets the user supplied username to be used while creating the PSCredential.
         /// </summary>
         [Parameter(Position = 0, Mandatory = false, ParameterSetName = messageSet)]
+        [ValidateNotNullOrEmpty]
```

Comment:
that is true, but that go with title and message also.
Ok so lets change them alle to
string title
if (this.MyInvocation.BoundParameters.ContainsKey("UserName")) 
{
username = this.UserName
}
else
{
title = Title
}
?